Description: Do not built python module with a soname and do not create symlinks
Author: Laurent Bigonville <bigon@debian.org>
Date: Tue, 13 Mar 2012 21:15:31 +0100

--- a/libqpol/swig/python/Makefile.am
+++ b/libqpol/swig/python/Makefile.am
@@ -1,4 +1,4 @@
-wrappedso_DATA = _qpol.so.@libqpol_version@
+wrappedso_DATA = _qpol.so
 wrappedso_SONAME = @libqpol_pyswig_soname@
 wrappedsodir = $(pkgpyexecdir)
 
@@ -12,22 +12,16 @@ BUILT_SOURCES = qpol_wrap.c
 AM_CFLAGS = @DEBUGCFLAGS@ @WARNCFLAGS@ @PROFILECFLAGS@ @SELINUX_CFLAGS@ \
 	-I$(top_builddir) -fpic \
 	-I$(top_srcdir)/libqpol/include
-AM_LDFLAGS = @DEBUGLDFLAGS@ @WARNLDFLAGS@ @PROFILELDFLAGS@ @QPOL_LIB_FLAG@ @PYTHON_LDFLAGS@
+AM_LDFLAGS = @DEBUGLDFLAGS@ @WARNLDFLAGS@ @PROFILELDFLAGS@ @QPOL_LIB_FLAG@
 
 $(BUILT_SOURCES): $(dist_noinst_DATA) $(DEPENDENCIES)
 	$(SWIG) $(SWIG_PYTHON_OPT) -o $@ $<
 
 $(wrappedso_DATA): $(BUILT_SOURCES)
-	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -Wl,-soname,$(wrappedso_SONAME)
-	$(LN_S) -f $@ $(wrappedso_SONAME)
-	$(LN_S) -f $@ _qpol.so
+	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
 
 $(wrappedpy_DATA): $(BUILT_SOURCES)
 
-install-data-hook:
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) $(wrappedso_SONAME)
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) _qpol.so
-
 uninstall-local:
 	-rm -rf $(DESTDIR)$(wrappedsodir)/$(wrappedso_SONAME) $(DESTDIR)$(wrappedsodir)/_qpol.so
 
--- a/libsefs/swig/python/Makefile.am
+++ b/libsefs/swig/python/Makefile.am
@@ -1,4 +1,4 @@
-wrappedso_DATA = _sefs.so.@libsefs_version@
+wrappedso_DATA = _sefs.so
 wrappedso_SONAME = @libsefs_pyswig_soname@
 wrappedsodir = $(pkgpyexecdir)
 
@@ -11,7 +11,7 @@ BUILT_SOURCES = sefs_wrap.cc
 AM_CXXFLAGS = @DEBUGCXXFLAGS@ @WARNCXXFLAGS@ @PROFILECFLAGS@ @SELINUX_CFLAGS@ \
 	@QPOL_CFLAGS@ @APOL_CFLAGS@ -I$(top_builddir) -fpic \
 	-I$(top_srcdir)/libsefs/include
-AM_LDFLAGS = @DEBUGLDFLAGS@ @WARNLDFLAGS@ @PROFILELDFLAGS@ @PYTHON_LDFLAGS@ \
+AM_LDFLAGS = @DEBUGLDFLAGS@ @WARNLDFLAGS@ @PROFILELDFLAGS@ \
 	@APOL_LIB_FLAG@ @QPOL_LIB_FLAG@ @SEFS_LIB_FLAG@ @XML_LIBS@
 DEPENDENCIES = $(top_builddir)/libqpol/src/libqpol.so \
 	$(top_builddir)/libapol/src/libapol.so \
@@ -23,16 +23,10 @@ $(BUILT_SOURCES): $(dist_noinst_DATA) $(
 	-I$(top_srcdir)/libqpol/swig -I$(top_srcdir)/libapol/swig $<
 
 $(wrappedso_DATA): $(BUILT_SOURCES)
-	$(CXX) -shared -o $@ $^ $(AM_CXXFLAGS) $(CXXFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -Wl,-soname,$(wrappedso_SONAME)
-	$(LN_S) -f $@ $(wrappedso_SONAME)
-	$(LN_S) -f $@ _sefs.so
+	$(CXX) -shared -o $@ $^ $(AM_CXXFLAGS) $(CXXFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
 
 $(wrappedpy_DATA): $(BUILT_SOURCES)
 
-install-data-hook:
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) $(wrappedso_SONAME)
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) _sefs.so
-
 uninstall-local:
 	-rm -rf $(DESTDIR)$(wrappedsodir)/$(wrappedso_SONAME) $(DESTDIR)$(wrappedsodir)/_sefs.so
 
--- a/libapol/swig/python/Makefile.am
+++ b/libapol/swig/python/Makefile.am
@@ -1,4 +1,4 @@
-wrappedso_DATA = _apol.so.@libapol_version@
+wrappedso_DATA = _apol.so
 wrappedso_SONAME = @libapol_pyswig_soname@
 wrappedsodir = $(pkgpyexecdir)
 
@@ -20,16 +20,10 @@ $(BUILT_SOURCES): $(dist_noinst_DATA) $(
 	$(SWIG) $(SWIG_PYTHON_OPT) -o $@ -I$(top_srcdir)/libapol/include -I$(top_srcdir)/libqpol/swig $<
 
 $(wrappedso_DATA): $(BUILT_SOURCES)
-	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -Wl,-soname,$(wrappedso_SONAME)
-	$(LN_S) -f $@ $(wrappedso_SONAME)
-	$(LN_S) -f $@ _apol.so
+	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
 
 $(wrappedpy_DATA): $(BUILT_SOURCES)
 
-install-data-hook:
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) $(wrappedso_SONAME)
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) _apol.so
-
 uninstall-local:
 	-rm -rf $(DESTDIR)$(wrappedsodir)/$(wrappedso_SONAME) $(DESTDIR)$(wrappedsodir)/_apol.so
 
--- a/libpoldiff/swig/python/Makefile.am
+++ b/libpoldiff/swig/python/Makefile.am
@@ -1,4 +1,4 @@
-wrappedso_DATA = _poldiff.so.@libpoldiff_version@
+wrappedso_DATA = _poldiff.so
 wrappedso_SONAME = @libpoldiff_pyswig_soname@
 wrappedsodir = $(pkgpyexecdir)
 
@@ -23,16 +23,10 @@ $(BUILT_SOURCES): $(dist_noinst_DATA) $(
 	-I$(top_srcdir)/libqpol/swig -I$(top_srcdir)/libapol/swig $<
 
 $(wrappedso_DATA): $(BUILT_SOURCES)
-	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -Wl,-soname,$(wrappedso_SONAME)
-	$(LN_S) -f $@ $(wrappedso_SONAME)
-	$(LN_S) -f $@ _poldiff.so
+	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
 
 $(wrappedpy_DATA): $(BUILT_SOURCES)
 
-install-data-hook:
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) $(wrappedso_SONAME)
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) _poldiff.so
-
 uninstall-local:
 	-rm -rf $(DESTDIR)$(wrappedsodir)/$(wrappedso_SONAME) $(DESTDIR)$(wrappedsodir)/_poldiff.so
 
--- a/libseaudit/swig/python/Makefile.am
+++ b/libseaudit/swig/python/Makefile.am
@@ -1,4 +1,4 @@
-wrappedso_DATA = _seaudit.so.@libseaudit_version@
+wrappedso_DATA = _seaudit.so
 wrappedso_SONAME = @libseaudit_pyswig_soname@
 wrappedsodir = $(pkgpyexecdir)
 
@@ -23,16 +23,10 @@ $(BUILT_SOURCES): $(dist_noinst_DATA) $(
 	-I$(top_srcdir)/libqpol/swig -I$(top_srcdir)/libapol/swig $<
 
 $(wrappedso_DATA): $(BUILT_SOURCES)
-	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -Wl,-soname,$(wrappedso_SONAME)
-	$(LN_S) -f $@ $(wrappedso_SONAME)
-	$(LN_S) -f $@ _seaudit.so
+	$(CC) -shared -o $@ $^ $(AM_CFLAGS) $(CFLAGS) $(SWIG_PYTHON_CPPFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
 
 $(wrappedpy_DATA): $(BUILT_SOURCES)
 
-install-data-hook:
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) $(wrappedso_SONAME)
-	cd $(DESTDIR)$(wrappedsodir) && $(LN_S) -f $(wrappedso_DATA) _seaudit.so
-
 uninstall-local:
 	-rm -rf $(DESTDIR)$(wrappedsodir)/$(wrappedso_SONAME) $(DESTDIR)$(wrappedsodir)/_seaudit.so
 
